### PR TITLE
Update wheels process to stage to gcs and pull version from branch

### PR DESCRIPTION
**Pull request by apilloud**
This updates the wheels build to stage artifacts to gcs. It also pulls versions from the branch.

### CODE REVIEW

**Comment by apilloud** [comment]
R: @aaltay @kennknowles 
The other half of this change is apache/beam#8315

**Review by kennknowles — APPROVED** [review]

File: .travis.yml [hunk 1]
```diff
@@ -91,8 +91,13 @@ install:
...
-after_success:
-    - ls -lh ${TRAVIS_BUILD_DIR}/wheelhouse/
-
-    - source deploy_travis.sh
-    - deploy
+deploy:
+  provider: gcs
+  access_key_id: ${ACCESS_KEY_ID}
+  secret_access_key: ${SECRET_ACCESS_KEY}
+  bucket: "beam-wheels-staging"
```
Comment (aaltay) [inline .travis.yml:98]: Is it possible to stage them under a directory with the version in the name?

**Comment by aaltay** [comment]
Merging this to accelerate RC5 if necessary. Please answer the open comment when you get a chance.

File: .travis.yml (hunk 1 above)
Comment (apilloud) [reply .travis.yml:98]: I looked into this when I wrote this change, there wasn't an obvious way to do this. Directory structures are copied as is.

**Pull request merged**